src/ucp/core/ucp_worker.c

```diff
@@ -1861,12 +1862,27 @@ ucp_worker_add_rkey_config(ucp_worker_h worker,
                                         sizeof(buf)));
     }
 
+    /* Save key-to-index lookup */
+    khiter = kh_put(ucp_worker_rkey_config, &worker->rkey_config_hash, *key,
+                    &khret);
+    if (khret == UCS_KH_PUT_FAILED) {
+        status = UCS_ERR_NO_MEMORY;
+        goto err;
+    }
+
+    /* We should not get into this function if key already exists */
+    ucs_assert_always(khret != UCS_KH_PUT_KEY_PRESENT);
+    kh_value(&worker->rkey_config_hash, khiter) = rkey_cfg_index;
+
     /* Initialize protocol selection */
     status = ucp_proto_select_init(&rkey_config->proto_select);
     if (status != UCS_OK) {
-        goto err;
+        goto err_kh_del;
     }
 
+    ++worker->rkey_config_count;
```

Comment:
moved here in case ucp_proto_select_short_init also adds rkey config (recursively)